test/rulesets/Base/base.jl

```diff
@@ -164,4 +176,22 @@
         frule_test(f, (x, Δx), (y, Δy), (z, Δz))
         rrule_test(f, Δk, (x, x̄), (y, ȳ), (z, z̄))
     end
+    @testset "trinary ($f)" for f in (clamp, )
+        x̄, ȳ, z̄    = randn(3)
+        Δx, Δy, Δz = randn(3)
+        Δk = randn()
+
+        x, y, z = 1., 2., 3.  # to left
+        frule_test(f, (x, Δx), (y, Δy), (z, Δz))
+        rrule_test(f, Δk, (x, x̄), (y, ȳ), (z, z̄))
+
+        x, y, z = 2.5, 2., 3.  # in the middle
+        frule_test(f, (x, Δx), (y, Δy), (z, Δz))
+        rrule_test(f, Δk, (x, x̄), (y, ȳ), (z, z̄))
+
+
+        x, y, z = 4., 2., 3.  # to right
+        frule_test(f, (x, Δx), (y, Δy), (z, Δz))
+        rrule_test(f, Δk, (x, x̄), (y, ȳ), (z, z̄))
+    end
```

Comment:
Since this test will never work for things other than clamp since it is handled the case about its boundries
would it be more clar to not use a `for` loop?

```suggestion
    @testset "clamp"  begin
        x̄, ȳ, z̄    = randn(3)
        Δx, Δy, Δz = randn(3)
        Δk = randn()

        x, y, z = 1., 2., 3.  # to left
        frule_test(clamp, (x, Δx), (y, Δy), (z, Δz))
        rrule_test(clamp, Δk, (x, x̄), (y, ȳ), (z, z̄))

        x, y, z = 2.5, 2., 3.  # in the middle
        frule_test(clamp, (x, Δx), (y, Δy), (z, Δz))
        rrule_test(clamp, Δk, (x, x̄), (y, ȳ), (z, z̄))

        x, y, z = 4., 2., 3.  # to right
        frule_test(clamp, (x, Δx), (y, Δy), (z, Δz))
        rrule_test(clamp, Δk, (x, x̄), (y, ȳ), (z, z̄))
    end
```